### PR TITLE
Add 3 tutorials for developers

### PR DESCRIPTION
**Pull request by xylar**
* Getting Started
* Understanding a task
* Adding a new task

### CODE REVIEW

**Comment by xylar** [comment]
@akturner, @alicebarthel, @anirban89, @cbegeman, @darincomeau, @jjbenedict, @katsmith133, @milenaveneziani 

Please let me know if you notice problems with any of the tutorials.  This would be an easy place to comment and fix things.

**Review by milenaveneziani — APPROVED** [review]
This looks great @xylar.

**Comment by darincomeau** [comment]
@xylar for the 'Add task' tutorial, an addition I'd recommend is indicate where the task needs to be added so that `mpas_analysis` will actually pick up and run the new task. For the sea ice production task I'm adding now, I found that I needed to add it to `mpas_analysis/sea_ice/__init__.py` and `mpas_analysis/__main__.py`.

**Comment by alicebarthel** [comment]
@darincomeau Thanks, your comment was really helpful.

**Comment by xylar** [comment]
@darincomeau, you're totally right, that's the missing step.  I will add that.

**Comment by xylar** [comment]
@darincomeau, I did add that, but I didn't update the temporary webpage on my github.io.  Take a look at this section and let me know if it addresses what you were missing:
https://github.com/MPAS-Dev/MPAS-Analysis/blob/66562cfee92ea73314243367975d96c8a8ea38cd/docs/tutorials/dev_add_task.rst#7-adding-the-task

**Comment by darincomeau** [comment]
@xylar that looks great - thanks!

**Comment by xylar** [comment]
There are some code blocks not showing up in the preview but which look right when I build the documentation.

**Review by darincomeau — APPROVED** [review]
Greatly appreciate your effort in putting together these tutorials!

File: docs/tutorials/dev_understand_a_task.rst [hunk 1]
```diff
@@ -0,0 +1,1193 @@
...
+                             'when climatology and ref year are the same.')
+
+In this particular case, we first call the super class' version of the
+:py:meth:`~mpas_analysis.shared.AnalysisTask.setup_and_check()` method.  This
+takes care of some important setup.
+
+Then, we use this method to check if the user has specified meaningful values
+for the climatology start and end year and the reference year.  If they happen
+to be the same, it doesn't really make sense to run the analysis and it will
+raise and error so the analysis gets skipped.
```
Comment (cbegeman) [inline docs/tutorials/dev_understand_a_task.rst:514]:
```suggestion
raise an error so the analysis gets skipped.
```

File: docs/tutorials/dev_understand_a_task.rst [hunk 2]
```diff
@@ -0,0 +1,1193 @@
...
+
+The ``setup_and_check()`` method of an analysis task is called when it is clear
+that this particular analysis has been requested (but before the analysis is
+actually ready to run).  This is in contrast to the constructor, which is
+run for *every* analysis task everytime MPAS-Analysis runs because we need
+information from the analysis task (its name, component and tags) in order to
+determine if it should run or not.
+
+In this method, we would typically perform checks to make sure the simulation
+has been configured properly ot run the analysis.  For example, is the
```
Comment (cbegeman) [inline docs/tutorials/dev_understand_a_task.rst:477]:
```suggestion
has been configured properly to run the analysis.  For example, is the
```

File: docs/tutorials/dev_understand_a_task.rst [hunk 3]
```diff
@@ -0,0 +1,1193 @@
...
+its "parent" task :py:class:`~mpas_analysis.ocean.ClimatologyMapOHCAnomaly`.
+It descends not from the :py:class:`~mpas_analysis.shared.AnalysisTask` base
+class but from another subtask,
+:py:class:`~mpas_analysis.shared.climatology.RemapMpasClimatologySubtask`.
+This tutorial won't attempt to cover
+:py:class:`~mpas_analysis.shared.climatology.RemapMpasClimatologySubtask` in
+all its detail.  The basics are that that class starts with MPAS climatology
+data over one or more ``seasons`` that has previously been computed by an
+:py:class:`~mpas_analysis.shared.climatology.MpasClimatologyTask` task.  It
+remaps that data form the MPAS mesh to one or more comparison grids (e.g.
```
Comment (cbegeman) [inline docs/tutorials/dev_understand_a_task.rst:533]:
```suggestion
remaps that data from the MPAS mesh to one or more comparison grids (e.g.
```

**Review by jjbenedict — APPROVED** [review]
I can only confirm that the getting_started section worked for me.  I have not done much with the other sections.

File: docs/tutorials/dev_add_task.rst [hunk 4]
```diff
@@ -0,0 +1,1405 @@
...
+
+4.2 Constructor
+~~~~~~~~~~~~~~~
+
+Then, I move on to the constructor.  The main things I need to do besides
+renaming the task are:
+
+* rename the field I'm processing to ``barotropicStreamfunction``.
+
+* clean up the ``tags`` a little bit (change ``anomaly`` ot ``streamfunction``).
```
Comment (cbegeman) [inline docs/tutorials/dev_add_task.rst:566]:
```suggestion
* clean up the ``tags`` a little bit (change ``anomaly`` to ``streamfunction``).
```

**Review by cbegeman — APPROVED** [review]
I noticed a few typos but overall the tutorials look great. Thanks!

**Comment by xylar** [comment]
@cbegeman, thanks for finding those typos! (I'm sure not the last.)  They should be fixed now.

**Review by akturner — APPROVED** [review]

File: docs/tutorials/dev_add_task.rst [hunk 5]
```diff
@@ -0,0 +1,1405 @@
...
+                                               nVertices))
+
+            solution = scipy.sparse.linalg.lsqr(M, rhs)
+
+            bsfVertex[tIndex, :] = -solution[0]
+
+        return bsfVertex
+
+
+    def _compute_barotropic_streamfunction_cell(dsMesh, bsfVertex):
```
Comment (anirban89) [inline docs/tutorials/dev_add_task.rst:193]: Here do we want to add the remapping to comparison grid?

File: docs/tutorials/dev_add_task.rst [hunk 6]
```diff
@@ -0,0 +1,1405 @@
...
+Then, I move on to the constructor.  The main things I need to do besides
+renaming the task are:
+
+* rename the field I'm processing to ``barotropicStreamfunction``.
+
+* clean up the ``tags`` a little bit (change ``anomaly`` to ``streamfunction``).
+
+* get rid of ``ref_year_climatology_task`` since I'm not computing anomalies.
+
+* get rid of ``depth_range`` because I'm using only the full ocean column.
```
Comment (anirban89) [inline docs/tutorials/dev_add_task.rst:570]: Do we add the depth range back in here?

**Review by anirban89 — COMMENTED** [review]
Do you want to add the updated task here in the tutorial? or leave the tutorial as is?

File: docs/tutorials/dev_add_task.rst (hunk 5 above)
Comment (xylar) [reply docs/tutorials/dev_add_task.rst:193]: If you are referring to the Subpolar North Atlantic, no, let's keep this simple.

File: docs/tutorials/dev_add_task.rst (hunk 6 above)
Comment (xylar) [reply docs/tutorials/dev_add_task.rst:570]: I don't think so.  Let's keep this as simple as possible for now.

**Comment by xylar** [comment]
@anirban89, if there are things I need to change for the barotropic streamfunciton example to make it work that would be great!  As far as adding complexity, I think we should leave that out of the tutorial.  It's already complicated enough without new projeciton grids or depth ranges.

**Comment by alicebarthel** [comment]
@xylar this looks great! I did not go through the last one in detail but read through it to use it as an example for my own task. Very helpful!

**Review by alicebarthel — APPROVED** [review]

**Review by anirban89 — APPROVED** [review]

**Comment by xylar** [comment]
Thank you everyone!  Very much appreciated.

**Pull request merged**